compose/internal/composebinary/composebinary.go

```diff
@@ -35,7 +35,12 @@ func NewComposeWrapper(binaryPath, configPath string) (libstack.Deployer, error)
 func (wrapper *ComposeWrapper) Deploy(ctx context.Context, workingDir, host, projectName string, filePaths []string, envFilePath string, forceRereate bool) error {
 	output, err := wrapper.Command(newUpCommand(filePaths, forceRereate), workingDir, host, projectName, envFilePath)
 	if len(output) != 0 {
-		log.Printf("[libstack,composebinary] [message: finish deploying] [output: %s] [err: %s]", output, err)
+		if err != nil {
+			log.Printf("[libstack,composebinary] [message: deploy complete] [output: %s] [err: %s]", output, err)
```

Comment:
any reason why you don't want the error to bubble up here? that condition looks weird:

If there is an error, log deploy complete?

I think it is not needed. If you need to handle the error, you should do it by catching the error when calling Deploy.

My comment applies to all the other occurence of this code change in this PR.

Reply:
oops, no that should be return err not return nil.  my bad.